src/helper.cc

```diff
@@ -848,20 +868,30 @@ helper::handleKilledServer(HelperServerBase *srv, bool &needsNewServers)
         debugs(84, DBG_CRITICAL, "WARNING: " << id_name << " #" << srv->index << " exited");
 
         if (childs.needNew() > 0) {
-            debugs(80, DBG_IMPORTANT, "Too few " << id_name << " processes are running (need " << childs.needNew() << "/" << childs.n_max << ")");
-
-            if (childs.n_active < childs.n_startup && last_restart > squid_curtime - 30) {
-                if (srv->stats.replies < 1)
-                    fatalf("The %s helpers are crashing too rapidly, need help!\n", id_name);
-                else
-                    debugs(80, DBG_CRITICAL, "ERROR: The " << id_name << " helpers are crashing too rapidly, need help!");
-            }
+            handleLackOfServers(srv->stats.replies >= 1);
```

Comment:
With the other requested change, this call should be moved up, right below the debugs() WARNING. This move will remove the (false) implication that the handleLackOfServers() caller must know _more_ than just the fact that we unexpectedly lost a server. In fact, I think we should _rename_ handleLackOfServers() to handleFewerServers() or similar to help squash that implication!

Just like the `flags.shutdown = true` bug discussed elsewhere, the handleFewerServers() call ought to be unconditional. The handling logic inside that method may change. The caller should not know about the current handling logic details.

Reply:
Done.